### PR TITLE
Add room members admin endpoint

### PR DESCRIPTION
**Pull request by awesome-michael**
Signed-off-by: Michael Albert <michael.albert@awesome-technologies.de>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: tests/rest/admin/test_room.py [hunk 1]
```diff
@@ -1136,6 +1136,50 @@ def test_single_room(self):
...
+        url = "/_synapse/admin/v1/rooms/%s/members" % (room_id_1,)
+        request, channel = self.make_request(
+            "GET", url.encode("ascii"), access_token=self.admin_user_tok,
+        )
+        self.render(request)
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+
+        self.assertIn("@foo:test", channel.json_body["members"])
+        self.assertIn("@bar:test", channel.json_body["members"])
+        self.assertEqual(channel.json_body["total"], 3)
```
Comment (clokep) [inline tests/rest/admin/test_room.py:1170]: Looks like these assertions would be clearer using [`assertCountEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual). (The name of that assertion is terrible, but pretty much it asserts that two lists have the same items in any order.)

File: docs/admin_api/rooms.md [hunk 2]
```diff
@@ -319,6 +319,33 @@ Response:
 }
 ```
 
+# Room Members API
+
+The Room Members admin API allows server admins to get a list of all members of a room.
```
Comment (clokep) [inline docs/admin_api/rooms.md:324]: Please add a description of the fields in the response.

File: synapse/rest/admin/rooms.py [hunk 3]
```diff
@@ -231,6 +231,30 @@ async def on_GET(self, request, room_id):
...
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        members = await self.store.get_users_in_room(room_id)
+        if not members:
+            raise NotFoundError("Room not found")
```
Comment (clokep) [inline synapse/rest/admin/rooms.py:251]: Couldn't this also be from everyone having left the room?

File: changelog.d/7842.feature [hunk 4]
```diff
@@ -0,0 +1 @@
+Add room members admin endpoint. Contributed by Awesome Technologies Innovationslabor GmbH.
```
Comment (clokep) [inline changelog.d/7842.feature:1]:
```suggestion
Add an admin API to list the users in a room. Contributed by Awesome Technologies Innovationslabor GmbH.
```

File: tests/rest/admin/test_room.py (hunk 1 above)
Comment (awesome-michael) [reply tests/rest/admin/test_room.py:1170]: Thanks for the hint, I was unsure about the order of the lists

File: synapse/rest/admin/rooms.py (hunk 3 above)
Comment (awesome-michael) [reply synapse/rest/admin/rooms.py]: Yes this is possible. I think these two are not distinguishable so I edited the error message.

File: synapse/rest/admin/rooms.py [hunk 5]
```diff
@@ -231,6 +231,31 @@ async def on_GET(self, request, room_id):
...
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        members = await self.store.get_users_in_room(room_id)
+        if not members:
+            print(members)
```
Comment (clokep) [inline synapse/rest/admin/rooms.py:251]: Looks like a `print` snuck in here!

Comment (awesome-michael) [reply synapse/rest/admin/rooms.py:251]: ah sorry... was testing if I could distinguish empty rooms from not found

File: synapse/rest/admin/rooms.py [hunk 6]
```diff
@@ -231,6 +231,30 @@ async def on_GET(self, request, room_id):
...
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        members = await self.store.get_users_in_room(room_id)
+        if not members:
+            raise NotFoundError("Room not found or empty")
```
Comment (clokep) [inline synapse/rest/admin/rooms.py:251]: After playing with this a bit more, I think this is going to be confusing, I think we should add use the `self.store.get_room(room_id)` method first, if that returns `None` we can raise `NotFoundError`.

Then, if there are no members I think we should return it as a success: `{"members": [], "total": 0}`.

Comment (awesome-michael) [reply synapse/rest/admin/rooms.py:251]: Thats a very good suggestion. I'll add the check.

**Pull request merged**